File: argentum-api/src/main/java/net/devium/argentum/rest/model/request/UserRequest.java
package net.devium.argentum.rest.model.request;

import net.devium.argentum.jpa.RoleEntity;
import net.devium.argentum.jpa.UserEntity;

import java.util.List;
import java.util.Set;

public class UserRequest {
    private Long id;
    private String username;
    private String password;
    private List<String> roles;

    public Long getId() {
        return id;
    }

    public String getUsername() {
        return username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public List<String> getRoles() {
        return roles;
    }

    // Doesn't check for changed password. Do so manually.
    public UserEntity toEntity(Set<RoleEntity> roles) {
        return new UserEntity(
                id != null ? id : -1,
                username,
                password,
                roles
        );
    }
}

File: argentum-ui/src/app/common/rest-service/request/user-request.ts
import { User } from '../../model/user';
export class UserRequest {
  id: number;
  username: string;
  password: string;
  roles: string[];
}

export function fromUser(user: User): UserRequest {
  return null;
  // return {
  //   id: user.id,
  //   username: user.username,
  //   password: user.password.length > 0 ? user.password : null,
  //   roles: user.roles
  // };
}
